### PR TITLE
Remove extra slash from path in 3rd-party funcs

### PR DESCRIPTION
**Pull request by castulo**
Signed-off-by: Castulo Martinez <castulo.martinez@intel.com>

### CODE REVIEW

**Comment by castulo** [comment]
@otaviobp take a look at this PR and let me know what you think. I'm unsure about two things:
1) the name of the function `sys_path`
2) if the function should live in sys.c or in helpers.c

File: src/3rd_party_repos.c [hunk 1]
```diff
@@ -34,27 +34,27 @@
 
 char *third_party_get_bin_dir(void)
 {
-	return sys_path_join(globals_bkp.path_prefix, SWUPD_3RD_PARTY_BIN_DIR);
+	return sys_path("%s%s", globals_bkp.path_prefix, SWUPD_3RD_PARTY_BIN_DIR);
```
Comment (otaviobp) [inline src/3rd_party_repos.c]: I would prefer to always use "%s/%s" to make sure we will never end up merging 2 paths without separators. sys_path is going to remove that anyway, so it's fine.

Comment (castulo) [reply src/3rd_party_repos.c]: good point

**Comment by castulo** [comment]
V2, addressed all the comments from review:
- replaced the current `sys_path_join` function with the new implementation that can join many paths
- fixed the arguments of the function everywhere where the function was already being used to match the new arguments

**Pull request merged**